### PR TITLE
Improve tracking parameters protection reference tests

### PR DESCRIPTION
**Pull request by kzar**
This change includes a number of improvements to the tracking
parameters protection reference tests:

 - Expand the tests to cover more edge-cases such as case-sensitivity of
   regular expression matching.
 - Name the test cases consistently.
 - Check allowlisted initiator domains and allowlisted subdomains are
   handled correctly.
 - Check that other edge-cases like parameter ordering and the
   handling of duplicate parameters are handled correctly.
 - Renamed example2.com to allowlisted.com to make the expected
   behaviour of the tests more obvious.

### CODE REVIEW

File: url-parameters/tests.json [hunk 1]
```diff
@@ -1,55 +1,124 @@
...
-                "name": "Remove utm tracking parameters from URL",
-                "testURL": "http://www.example.com/test.html?utm_source=test&utm_medium=test&utm_campaign=test&utm_term=test&utm_content=test",
-                "expectURL": "http://www.example.com/test.html",
+                "name": "Non-tracking parameters are not removed",
+                "testURL": "http://www.example.com/test.html?q=test&id=5",
+                "expectURL": "http://www.example.com/test.html?q=test&id=5",
+                "exceptPlatforms": []
+            },
+            {
+                "name": "Parameters with values that match tracking parameter names are not removed",
```
Comment (kdzwinel) [inline url-parameters/tests.json:14]: nice!

File: url-parameters/tests.json [hunk 2]
```diff
@@ -1,55 +1,124 @@
...
+                "name": "Parameters similar to plain-text tracking parameters are not removed",
+                "testURL": "http://www.example.com/test.html?fbclid1=test",
+                "expectURL": "http://www.example.com/test.html?fbclid1=test",
                 "exceptPlatforms": []
             },
             {
-                "name": "Don't remove parameters because domain is in exceptions list",
-                "testURL": "http://example2.com/test.html?gclid=test&fbclid=test",
-                "expectURL": "http://example2.com/test.html?gclid=test&fbclid=test",
+                "name": "Parameters case insensitively matching plain-text tracking parameters are not removed",
```
Comment (kdzwinel) [inline url-parameters/tests.json:38]: oh, that's interesting, I'd expect them to be removed I think. @SlayterDev what's your take? Is that specified in the spec?

**Review by kdzwinel — APPROVED** [review]
Great work tracking down all those edge cases👏 Are all of those covered by the spec?

It's a thumbs up from me, but @SlayterDev should also take a look.

File: url-parameters/tests.json (hunk 2 above)
Comment (kzar) [reply url-parameters/tests.json:38]: I figured case-sensitive matching was implied by the `"utm_[a-zA-Z]*"` trackingParameters entry in extension-config.json. FWIW I also think that is the right call, since case-sensitive matching could turn out to be useful in the future.

Comment (kzar) [reply url-parameters/tests.json:38]: Also, if we do plain-text parameter matching case-sensitively that matches how the declarativeNetRequest API's QueryTransform conditions work. That will make porting the feature to mv3 easier.

Comment (SlayterDev) [reply url-parameters/tests.json:38]: We can imply case sensitivity. FWIW I've never seen a parameter that isn't just lowercase letters.

**Review by SlayterDev — APPROVED** [review]
LGTM! 👍
